intern/cycles/kernel/kernel_types.h

```diff
@@ -247,7 +247,7 @@ enum PathTraceDimension {
   PRNG_LENS_U = 2,
   PRNG_LENS_V = 3,
   PRNG_TIME = 4,
-  PRNG_UNUSED_0 = 5,
+  PRNG_WAVELENGTH  = 5,
```

Comment:
Are you familiar with RNG for sampling, and why it is different to 'regular' RNG? If you do, feel free to skip this. If not, here's a quick intro:

For a monte-carlo renderer, you're trying to guess what the integral of an infinite dimension equation is (example dimensions being things like 'where within the pixel am I sampling', or 'what direction did this ray bounce off this material' or 'what point in time of the render am I simulating'). You do this by sampling points in the high dimensional space. You could do this in a regular way (start at [0.1, 0, 0, 0, ...] and just count up incrementally) but that gives you some nasty correlation issues (like soft shadows not being as soft as you might like).

You _could_ use purely random numbers, and that does work, but let's say two of your numbers land right beside each other, you've essentially wasted those calculations because you could have used the second sample to find out the result of the equation at a different point.

So, there's an inbetween approach - generate numbers which don't seem to relate to each other in any easy pattern, but make sure that the samples you make are spread as evenly as possible over the sample space.

The reason we have a different seed for each dimension here is so that we don't get correlation issues like 'whenever I'm sampling the start of the frame in time, I'm also sampling long wavelengths' (although now I think about it that sounds really cool and I want to try it out 😄 )